fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1421,18 +1430,25 @@ ACTOR Future<GetReadVersionReply> getLiveCommittedVersion(Span parentSpan, Proxy
 		g_traceBatch.addEvent("TransactionDebug", debugID.get().first(), "MasterProxyServer.getLiveCommittedVersion.confirmEpochLive");
 	}
 
-	vector<GetReadVersionReply> versions = wait(getAll(proxyVersions));
-	GetReadVersionReply rep;
-	rep.version = commitData->committedVersion.get();
+	state GetReadVersionReply rep;
 	rep.locked = commitData->locked;
 	rep.metadataVersion = commitData->metadataVersion;
-	rep.recentRequests = commitData->stats.getRecentRequests();
+	rep.version = commitData->committedVersion.get();
 
-	for (auto v : versions) {
-		if(v.version > rep.version) {
-			rep = v;
+	if (SERVER_KNOBS->ASK_READ_VERSION_FROM_MASTER) {
+		GetReadVersionReply replyFromMaster = wait(replyFromMasterFuture);
+		if (replyFromMaster.version > rep.version) {
```

Comment:
always send the reply from the master

Reply:
When we serve the first GRV, the version in the master's reply will be the same as proxy's committedVersion which is recoveryTransactionVersion. But at this time, the database maybe already locked, it can be reflected by proxy's locked value but not reflected in master's locked value. So we use master's reply only when its version is larger than the proxy's version which means another proxy has reported a larger value and master's locked value reflects that proxy's locked value.

Reply:
I though Evan is asking:
forward the rep to master and let master to use the rep to send the result to client.
The traffic flow is Client -> Proxy -> Master -> Client.

In the current implementation, the traffic flow is Client -> Proxy -> **Master -> Proxy** -> Client. It has one more hop compared to what Evan suggested. 

The master can still compare with the rep.version to decide which version to use.